### PR TITLE
Enable the use of py_reader in PaddingRNN

### PR DESCRIPTION
**Pull request by Xreki**

### CODE REVIEW

File: PaddingRNN/lstm_paddle/train.py [hunk 1]
```diff
@@ -174,40 +174,64 @@ def main():
...
+                loss, last_hidden, last_cell, feed_order = lm_model.lm_model(
+                    hidden_size,
+                    vocab_size,
+                    batch_size,
+                    num_layers=num_layers,
+                    num_steps=num_steps,
+                    init_scale=init_scale,
+                    dropout=dropout, 
+                    rnn_model=rnn_model,
+                    use_py_reader=False)
```
Comment (chengduoZH) [inline PaddingRNN/lstm_paddle/train.py]: The above code should be polished, for example, 
```
vars = lm_model.lm_model(
                    hidden_size,
                    vocab_size,
                    batch_size,
                    num_layers=num_layers,
                    num_steps=num_steps,
                    init_scale=init_scale,
                    dropout=dropout, 
                    rnn_model=rnn_model,
                    use_py_reader=args.use_py_reader)
if args.use_py_reader:
      py_reader = vars[-1]
      vars = []
loss, last_hidden, last_cell, feed_order = vars
```

File: PaddingRNN/lstm_paddle/train.py [hunk 2]
```diff
@@ -411,13 +444,88 @@ def train_an_epoch(epoch_id, batch_times):
...
+
+        init_hidden, init_cell = get_init_data()
+        first_data_feeds = {}
+        first_data_feeds["init_hidden"] = init_hidden
+        first_data_feeds["init_cell"] = init_cell
+
+        total_loss = 0
+        iters = 0
+
+        py_reader.start()
```
Comment (chengduoZH) [inline PaddingRNN/lstm_paddle/train.py]: 这里应该指定跑几个epoch

File: PaddingRNN/lstm_paddle/train.py [hunk 3]
```diff
@@ -411,13 +444,88 @@ def train_an_epoch(epoch_id, batch_times):
         ppl = np.exp(total_loss / iters)
         return ppl
 
+    def train_an_epoch_py_reader(epoch_id, batch_times):
+        data_iter_size = batch_size
+        if device_count > 1 and args.parallel:
+            data_iter_size = batch_size * device_count
```
Comment (chengduoZH) [inline PaddingRNN/lstm_paddle/train.py]: data_iter_size = batch_size / device_count

File: PaddingRNN/lstm_paddle/train.py [hunk 4]
```diff
@@ -411,13 +444,88 @@ def train_an_epoch(epoch_id, batch_times):
...
+                        fetch_list=[loss.name, "learning_rate"],
+                        use_program_cache=True)
+                else:
+                    batch_time = time.time() - batch_start_time
+                    batch_times.append(batch_time)
+                    batch_start_time = time.time()
+                    fetch_outs = exe.run(
+                        train_program,
+                        fetch_list=[loss.name, "learning_rate"],
+                        use_program_cache=True)
```
Comment (chengduoZH) [inline PaddingRNN/lstm_paddle/train.py]:
```
feed = None
if batch_id == 0:
      feed=first_data_feeds
batch_time = time.time() - batch_start_time
batch_times.append(batch_time)
batch_start_time = time.time()
fetch_outs = exe.run(
                        train_program,
                        feed = feed,
                        fetch_list=[loss.name, "learning_rate"],
                        use_program_cache=True)
```

File: PaddingRNN/lstm_paddle/train.py (hunk 1 above)
Comment (Xreki) [reply PaddingRNN/lstm_paddle/train.py]: Done.

File: PaddingRNN/lstm_paddle/train.py (hunk 3 above)
Comment (Xreki) [reply PaddingRNN/lstm_paddle/train.py]: Done.

File: PaddingRNN/lstm_paddle/train.py (hunk 4 above)
Comment (Xreki) [reply PaddingRNN/lstm_paddle/train.py]: Done.

**Pull request merged**